### ISSUE TITLE
Query references with additional filtering & ordering

### ISSUE THREAD

**Issue by eoconnor** [JetBrains/Exposed#1590]
I'm using the DAO API of Exposed, and I have a question about querying references. Let's say I have these two tables:
```
class User(id: EntityID<Int>): IntEntity(id) {
    companion object : IntEntityClass<User>(Users)
    var name by Users.name
}
class UserRating(id: EntityID<Int>): IntEntity(id) {
    companion object : IntEntityClass<UserRating>(UserRatings)
    var value by UserRatings.value
    var timestamp by UserRatings.timestamp
    var user by User referencedOn UserRatings.user
}
```
And let's further say that I want to query for all a user's ratings, _but_ I also want the query to filter by results where the `UserRating.value` is above a certain threshold, _and_ I want to order the query results by the `UserRating.timestamp` value.

Does Exposed support that?

**Comment by AlexeySoshin** [comment]
Don't see why not?

Try something like:
```kotlin
val threshold = 3
UserRating.find { UserRating.value greater threshold }.sortedBy { UserRating.timestamp }
```

You can take a look at the DAO documentation for more information:
https://github.com/JetBrains/Exposed/wiki/DAO#read

**Comment by eoconnor** [comment]
Yes, but that doesn't filter by user. The DAO docs talk about adding this to the User entity class to make it easier to look up references:
```
val ratings by UserRating referrersOn UserRatings.film
```
So then you can just do this:
```
user.ratings
```
My question was how to _combine_ that with other query filtering.

**Comment by AlexeySoshin** [comment]
Sorry, I misunderstood your question.

You want to do something like:
user.ratings.find { it.value > threshold }

And that it would produce a query.

From what I can see, this is isn't possible.
That code would load all ratings into memory, create a Kotlin collection, then filter that collection (sort it, etc).


You'll have to resort to something like:
```kotlin
val res = UserRating.find { (UserRatings.value greater threshold) and (UserRatings.user eq 1) }
            .sortedBy { it.timestamp }
            ```
